### PR TITLE
WT-18147 Fix wt tool's pretty-print support for zero-length "u" data

### PR DESCRIPTION
**Pull request by DylanSHLiang**
Reland of #14485 (reverted in #14566). No code changes from the original PR.

`wt dump -p` printed a record's **previous** value instead of a blank value for a zero-length `u`/`U`-format field, because `__wt_buf_set_printable_format()` reused an output buffer and returned early without writing (or NUL-terminating) anything for the zero-length case.

`src/support/scratch.c`: the zero-length `'U'`/`'u'` case now writes the field separator (still nothing else) so the buffer is refreshed instead of left stale. Output is blank, matching `-x`, `-j`.

`test/suite/test_dump06.py` asserts the pretty dump shows blank.

Original revert reason: this broke mongo's `jstests/replsets/container_operations.js`, which expected `"(null)"` for empty values, and `wt_file_helper.js`'s `wtExtractRecordsFromDump` filtered out blank lines. Those need a mongo-side fix to accept blank output before/alongside this landing again.

Changes approved in https://github.com/10gen/mongo/pull/62291

### CODE REVIEW

File: test/suite/test_dump06.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
```
Comment (whoisong) [inline test/suite/test_dump06.py:1]: Still same problem like previous ones, do we really need a full new test suite to support this `u`? If it only covers a subset of edge case, can we simply extend an existing test_dump0{n} for it

File: src/support/scratch.c [hunk 2]
```diff
@@ -173,8 +173,15 @@ __wt_buf_set_printable_format(WT_SESSION_IMPL *session, const void *buffer, size
...
         case 'U':
         case 'u':
-            if (pv.u.item.size == 0)
+            if (pv.u.item.size == 0) {
+                /*
+                 * The buffer is reused across records; without writing anything here, buf->data
+                 * would still reference the previous record's formatted string instead of an empty
+                 * one.
+                 */
+                WT_ERR(__wt_buf_catfmt(session, buf, "%s", sep));
```
Comment (whoisong) [inline src/support/scratch.c:182]: So here seems sep is not reset to `,`, and where is `sep` comes from? What is this line for? if we keep sep as empty, then catfmt means append nothing to buf, didn't get what's the target here, the comment said still reference previous, but catfmt is keep previous string and append an empty

File: test/suite/test_dump06.py (hunk 1 above)
Comment (DylanSHLiang) [reply test/suite/test_dump06.py:1]: oh wait, sorry, this should have just been the same as the previous PR

Comment (DylanSHLiang) [reply test/suite/test_dump06.py:1]: nvm, I was thinking of a different PR.

Comment (Sean04) [reply test/suite/test_dump06.py:1]: agreed with Albert

File: src/support/scratch.c (hunk 2 above)
Comment (DylanSHLiang) [reply src/support/scratch.c]: I've moved the `sep` to the beginning of the loop in [239caa3](https://github.com/wiredtiger/wiredtiger/pull/14664/commits/239caa3c3c0bc7bbc56c236155b2439d69a58d75). The reason I need to call `__wt_buf_catfmt` is that the buffer is not reset across each call of the function. I was combining the resetting and adding the separator `,` into one which was confusing. This new approach does these two steps one at a time and is clearer.

File: src/support/scratch.c [hunk 3]
```diff
@@ -158,36 +158,46 @@ __wt_buf_set_printable_format(WT_SESSION_IMPL *session, const void *buffer, size
     p = (const uint8_t *)buffer;
     end = p + size;
 
+    /*
+     * The buffer is reused across calls and __wt_buf_init only resets its size, it neither clears
+     * nor NUL-terminates the existing contents. Write an empty string to reset the buffer.
+     */
     WT_ERR(__wt_buf_init(session, buf, 0));
+    WT_ERR(__wt_buf_catfmt(session, buf, "%s", ""));
```
Comment (whoisong) [inline src/support/scratch.c]: If the buf need to be inited to an empty string not `NULL`, use `__wt_buf_fmt` instead.

**Review by whoisong — APPROVED** [review]
One suggestion, other's good.

File: src/support/scratch.c (hunk 3 above)
Comment (DylanSHLiang) [reply src/support/scratch.c]: done in [33b72df](https://github.com/wiredtiger/wiredtiger/pull/14664/commits/33b72dfcbf35c508825ae677a048c2ac06b8f964). Removed `__wt_buf_init` since `__wt_buf_fmt` does init work as well

**Review by Sean04 — APPROVED** [review]
lgtm
